File: js/3020.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const [N, H] = inputs.shift().split(' ').map(Number);
const bottom = Array(H + 1).fill(0);
const top = Array(H + 1).fill(0);

inputs.forEach((e, i) => {
  if (i % 2 === 0) bottom[+e]++;
  else top[H - e + 1]++;
});

for (let i = 1; i <= H; i++) {
  top[i] += top[i - 1];
  bottom[H - i] += bottom[H - i + 1];
}

let min = Infinity;
let cnt = 0;
for (let i = 1; i <= H; i++) {
  if (top[i] + bottom[i] < min) {
    min = top[i] + bottom[i];
    cnt = 1;
  } else if (top[i] + bottom[i] === min) cnt++;
}

console.log(min, cnt);

File: c++/3020.cpp
#include <iostream>

using namespace std;

int top[500001];
int bottom[500001];

int main() {
  freopen("example.txt", "r", stdin);
  int N, H;
  cin >> N >> H;

  for (int i = 0; i < N; i++) {
    int h;
    cin >> h;
    if (i % 2 == 0) {
      bottom[h]++;
    } else {
      top[H - h + 1]++;
    }
  }

  for (int i = 1; i <= H; i++) {
    bottom[H - i] += bottom[H - i + 1];
    top[i] += top[i - 1];
  }

  int min = 1000000;
  int cnt = 0;
  for (int i = 1; i <= H; i++) {
    if (bottom[i] + top[i] < min) {
      min = bottom[i] + top[i];
      cnt = 1;
    } else if (bottom[i] + top[i] == min) {
      cnt++;
    }
  }

  cout << min << ' ' << cnt;
  return 0;
}